config/config/src/index.ts

```diff
@@ -257,14 +262,16 @@ export async function getConfig (opts: {
     ? pnpmConfig.rawLocalConfig['user-agent']
     : `${packageManager.name}/${packageManager.version} npm/? node/${process.version} ${process.platform} ${process.arch}`
   pnpmConfig.rawConfig = Object.assign.apply(Object, [
-    {
-      registry: 'https://registry.npmjs.org/',
-      '@jsr:registry': 'https://npm.jsr.io/',
-    },
+    {},
     ...[...npmConfig.list].reverse(),
     cliOptions,
     { 'user-agent': pnpmConfig.userAgent },
   ] as any) // eslint-disable-line @typescript-eslint/no-explicit-any
+  for (const key in pnpmConfig.rawConfig) {
+    if (!isRcSetting(key, opts.rcOptionsTypes)) {
+      delete pnpmConfig.rawConfig[key]
+    }
+  }
```

Comment:
There's a problem I'm trying to solve: I've found that the `pnpm config` CLI was pretty divorced from the actual configuration used by other parts of pnpm, for example: `pnpm config list` would list workspace-only settings from an `rc` file despite not actually using them in `pnpm install`. This is because `pnpm config` CLI reads from `rawConfig`.

I also want it to be only 1 source of truth, which would be `rawConfig`. The source of truth shouldn't include unknown settings.